[LWS49] iP

LWS49 opened the pull request:
### Luke

> Luke, your simple to-do list simpler than Notion.

Luke is your CLI checklist that helps you to keep track of all your tasks, big and small. It's,

- text-based
- easy to learn
- very convenient

All you need to do is,

1. download it from [here](https://www.google.com/).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉


And it is FREE!

Features:

- [X] Managing tasks
- [X] Edit tasks
- [ ] Reminders

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

LWS49 committed:
complete Level-0

LWS49 committed:
complete Level-1

LWS49 committed:
complete Level-1

LWS49 committed:
added Level-3

LWS49 committed:
added Level-4

LWS49 committed:
update javac and java

LWS49 committed:
remove default case without command

LWS49 committed:
commit to push Level-4 to master

LWS49 committed:
added Level-5

LWS49 committed:
added Level-5

LWS49 committed:
added Level-6

LWS49 committed:
Merge branch 'master' into push-Level-4

LWS49 committed:
Merge pull request #1 from LWS49/push-Level-4

Push level 4

LWS49 committed:
added Level-7

LWS49 committed:
added Level-8

LWS49 committed:
clean up code based on code analysis

LWS49 committed:
Merge pull request #2 from LWS49/branch-Level-8

Merging branch level 8 to origin

LWS49 committed:
Revert "Merging branch level 8 to origin"

LWS49 committed:
Merge pull request #3 from LWS49/revert-2-branch-Level-8

Revert "Merging branch level 8 to origin"

LWS49 committed:
Merge branch 'master' into branch-Level-8

LWS49 committed:
added Level-8

LWS49 committed:
Merge branch 'branch-Level-8'

aureliony commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+    protected DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("M/dd/yyyy HHmm");
+
+    protected DateTimeFormatter dateTimeStringFormatter = DateTimeFormatter.ofPattern("dd MMM uuuu hh:mma");
+    public Deadline(String description, String by) {
```
Should there be a blank line before this line? Should there be a blank line after it?

LWS49 committed:
completed core of Luke, writing OOP

aureliony commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,246 @@
+
+public class Luke {
+
+    static String[] validCommands = {"bye", "list", "unmark", "mark", "todo", "event", "deadline", "delete"};
+
+    static String directoryPath = "./data";
+
+    static String fileName = "list.txt";
+
+    static File file = new File (directoryPath, fileName);
```
Style: unnecessary space between File and (

aureliony replied:
Missing visiblity for these variables. Should they also be declared as constants?
Should there be blank lines between each declaration?

aureliony commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,246 @@
+            Scanner myReader = new Scanner(file);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                if (data != "") {
```
Should we use `!=` for comparing strings in java?

aureliony commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,246 @@
+
+        System.out.println("Hello! I'm " + name + "\nWhat can I do for you?");
+
+        Scanner scanner = new Scanner (System.in);
```
Style: unnecessary space

aureliony commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,52 @@
+
+    protected LocalDateTime toDate;
+    protected boolean hasToDate;
+    protected DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("M/dd/yyyy HHmm");
+
+    protected DateTimeFormatter dateTimeStringFormatter = DateTimeFormatter.ofPattern("dd MMM uuuu hh:mma");
```
Should there be blank lines between each declaration?

Xuyan0518 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,246 @@
+
+    static boolean fileIsEmpty = true;
+
+    private static boolean isCommandValid(String command) {
```
Correct naming for the method that returns a boolean. Nice!

Xuyan0518 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected boolean done;
```
Wrong form of naming a boolean value though. Probably should name it to isDone?

Xuyan0518 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,246 @@
+        Scanner scanner = new Scanner (System.in);
+        String input = scanner.nextLine();
+        while (!input.equals("bye")) {
+            String[] splited = input.split(" ");
```
Perhaps can describe it a bit more rather than using splited as a variable name? (by the way, it should be split).

Xuyan0518 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,246 @@
+                   taskList.add(deadline);
+                   break;
+               case "E":
+                   String[] eventFirstSplit = taskString.split("from: ");
```
Maybe can describe what is the product after the first split and second split instead of naming them as eventFirstSplit and eventSecondSplit.

Xuyan0518 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,246 @@
+            } catch (LukeException e) {
+                System.out.println(e);
+            }
+            switch (command) {
```
Wrong indent for switch statements.

Xuyan0518 reviewed (commented):
Overall good job. Just a few small changes to the naming of variables.

LWS49 committed:
finish OOP

LWS49 committed:
Merge branch 'branch-A-MoreOOP'

LWS49 committed:
finish A-Packages

LWS49 committed:
Merge branch 'branch-A-Packages'

LWS49 committed:
finish A-Gradle

LWS49 committed:
Merge branch 'branch-A-Gradle'

LWS49 committed:
finish A-JUnit

LWS49 committed:
move tests to correct package

LWS49 committed:
Merge branch 'branch-A-JUnit'

LWS49 committed:
completed A-JavaDoc

LWS49 committed:
completed A-CodingStandard

LWS49 committed:
complete A-Find

LWS49 committed:
Merge branch 'branch-A-JavaDoc'

LWS49 committed:
resolve merge conflicts between master and A-CodingStandards

LWS49 committed:
correct JavaDocs error

LWS49 committed:
correct JavaDocs error

LWS49 committed:
Revert "correct JavaDocs error"

This reverts commit 021cd8ee3008ddbeca4c9ec40e27dcd48ca08b3f.

LWS49 committed:
add back files that are accidentally deleted

LWS49 committed:
add back files that are accidentally deleted

LWS49 committed:
add back Luke

LWS49 committed:
add back Ui

LWS49 committed:
resolve merge conflicts between master and branch-Level-9

LWS49 committed:
remove duplicate case in handleInput()

LWS49 committed:
completed A-CheckStyle

LWS49 committed:
Merge branch 'branch-A-CheckStyle'

LWS49 committed:
add GUI components as stated in website

LWS49 committed:
completed Level-10

LWS49 committed:
Merge branch 'branch-Level-10'

LWS49 committed:
add assertions for Task subclass methods

Task subclass methods like changeBy, changeTo and changeFrom requires
them to be called by subclasses like Deadline and Event. Assertions are
added to ensure these methods are not accidentally called by the Task
class, without making them abstract (as each method should only be
called by one subclass - making it abstract and duplicating all unnecessary subclass methods is inefficent).

LWS49 committed:
Merge branch 'branch-A-Assertions'

LWS49 committed:
convert field variables to local variables

Many of the field variables are only called once and are recalled within
the same class. Thus, they are converted to local variables to keep the
code clean.

LWS49 committed:
complete A-CodeQuality

LWS49 committed:
Merge branch 'branch-A-CodeQuality'

LWS49 committed:
complete C-Update

LWS49 committed:
Merge branch 'branch-C-Update'

LWS49 committed:
complete checkstyle and build

LWS49 committed:
modify README and add picture

LWS49 committed:
Merge branch 'branch-A-UserGuide'

LWS49 committed:
add link to download

LWS49 committed:
update filepath

LWS49 committed:
update gitignore and README

LWS49 committed:
Remove luke.jar from remote repository access

LWS49 committed:
updated tests to be more complete

LWS49 committed:
make "bye" exit the app
